src/java.base/share/classes/javax/security/auth/Subject.java

```diff
@@ -319,6 +329,181 @@ public Subject run() {
         });
     }
 
+
+    private static enum SubjectStorage {
+        ACC, THREAD_LOCAL, SCOPE_LOCAL
+    };
+
+    private static final SubjectStorage SUBJECT_STORAGE
+            = SharedSecrets.getJavaLangAccess().systemAllowSecurityManager()
+                    ? SubjectStorage.ACC
+                    : SubjectStorage.THREAD_LOCAL;
+
+//    private static final ScopeLocal<Subject> SUBJECT_SCOPE
+//            = ScopeLocal.inheritableForType(Subject.class);
+
+    private static final InheritableThreadLocal<Subject> SUBJECT_THREAD_LOCAL =
+            new InheritableThreadLocal<Subject>() {
+                @Override protected Subject initialValue() {
+                    return null;
+                }
+            };
+
+    /**
+     * Return the current installed subject.
```

Comment:
Nit. I know other accessor methods in Subject use the word "Return", but I think "Returns" is more grammatically correct and reads better when describing what this method does. Also just say "Returns the current subject."

Reply:
I'll use "returns".